pyccel/codegen/printing/ccode.py

```diff
@@ -1478,7 +1471,26 @@ def _print_IndexedElement(self, expr):
                     inds[i] = IfTernaryOperator(PyccelLt(ind, LiteralInteger(0)),
                         PyccelAdd(base_shape[i], ind, simplify = True), ind)
 
+        if isinstance(base.class_type, HomogeneousListType):
+            assign = expr.get_user_nodes(Assign)
+            index = self._print(inds[0])
+            list_var = self._print(ObjectAddress(base))
+            container_type = self.get_c_type(base.class_type)
+            if assign:
+                assert len(assign) == 1
+                assign_node = assign[0]
+                if assign_node is expr or assign_node.lhs.is_user_of(expr):
```

Comment:
This does not seem to catch when `_at_mut` should be used

Reply:
Oops. Fixed [5ee6bba](https://github.com/pyccel/pyccel/pull/1926/commits/5ee6bba522b615ca043bfc5bbca850fe3dd88952)

I will add a set of tests for this